src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -254,24 +254,19 @@ fn check_place(
     place: &Place<'tcx>,
     span: Span,
 ) -> McfResult {
-    match place {
-        Place::Base(PlaceBase::Local(_)) => Ok(()),
-        // promoteds are always fine, they are essentially constants
-        Place::Base(PlaceBase::Static(box Static { kind: StaticKind::Promoted(_), .. })) => Ok(()),
-        Place::Base(PlaceBase::Static(box Static { kind: StaticKind::Static(_), .. })) =>
-            Err((span, "cannot access `static` items in const fn".into())),
-        Place::Projection(proj) => {
-            match proj.elem {
-                | ProjectionElem::ConstantIndex { .. } | ProjectionElem::Subslice { .. }
-                | ProjectionElem::Deref | ProjectionElem::Field(..) | ProjectionElem::Index(_) => {
-                    check_place(&proj.base, span)
-                }
-                | ProjectionElem::Downcast(..) => {
-                    Err((span, "`match` or `if let` in `const fn` is unstable".into()))
-                }
+    place.iterate(|place_base, place_projection| {
+        for proj in place_projection {
+            if let ProjectionElem::Downcast(..) = proj.elem {
```

Comment:
please keep the exhaustive matches (even if the arms end up just being `{}`), they are there to guard against future changes to `ProjectionElem`